pkg/controller/component/component_controller_test.go

```diff
@@ -123,6 +130,100 @@ func TestComponentController(t *testing.T) {
 		require.Equal(t, Name+":latest", dc.Spec.Triggers[1].ImageChangeParams.From.Name, "deployment config should be triggered by DeploymentTriggerOnImageChange from bc-output")
 		require.Equal(t, 1, len(dc.Labels), "dc should contain one label")
 		require.Equal(t, Name, dc.ObjectMeta.Labels["app"], "dc should have one label with name of CR.")
+
+		svc := &corev1.Service{}
+		errGetSvc := cl.Get(context.Background(), types.NamespacedName{Namespace: Namespace, Name: Name}, svc)
+		require.NoError(t, errGetSvc, "service is not created")
+		require.Equal(t, 1, len(svc.Spec.Ports), "service is using one port")
+		require.Equal(t, int32(8080), svc.Spec.Ports[0].Port, "default service port should be 8080")
+	})
+
+	t.Run("with ReconcileComponent CR containing all optional fields for service port and route should create resources", func(t *testing.T) {
+		//given
+		// Objects to track in the fake client.
+		// A Component resource with metadata and spec.
+		cpOptional := &compv1alpha1.Component{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      Name,
+				Namespace: Namespace,
+			},
+			Spec: compv1alpha1.ComponentSpec{
+				BuildType: "nodejs",
+				Codebase:  "https://somegit.con/myrepo",
+				Port:      "3000",
+				Exposed:   true,
+			},
+		}
+		objs := []runtime.Object{
+			cpOptional,
+		}
+		// Create a fake client to mock API calls.
+		cl := fake.NewFakeClient(objs...)
+
+		// Create a ReconcileComponent object with the scheme and fake client.
+		r := &ReconcileComponent{client: cl, scheme: s}
+
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      Name,
+				Namespace: Namespace,
+			},
+		}
+
+		//when
+		_, err := r.Reconcile(req)
+
+		//then
+		require.NoError(t, err, "reconcile is failing")
+
+		svc := &corev1.Service{}
+		errGetSvc := cl.Get(context.Background(), types.NamespacedName{Namespace: Namespace, Name: Name}, svc)
+		require.NoError(t, errGetSvc, "service is not created")
+		require.Equal(t, 1, len(svc.Spec.Ports), "service is using one port")
+		require.Equal(t, int32(3000), svc.Spec.Ports[0].Port, "service port should be 3000")
```

Comment:
I would also verify endpoint for that service is created. you can find that using `oc get ep`